From 9c97bd02042d37e3654246b2edd2cf0e0b16ddfc Mon Sep 17 00:00:00 2001
From: Vijay Khemka <axiado1321@axiado.com>
Date: Tue, 21 Mar 2023 00:29:58 -0700
Subject: [PATCH 1/2] Add meson file for Axiado yaml

Added Axiado in meson target and also added all meson files
generated by Axiado yaml interface file. Actual yaml file is
in the next patch

Signed-off-by: Vijay Khemka <axiado1321@axiado.com>
---
 gen/com/axiado/ax200/Demo/meson.build | 14 ++++++++++++++
 gen/com/axiado/ax200/meson.build      | 15 +++++++++++++++
 gen/com/axiado/meson.build            |  2 ++
 gen/com/meson.build                   |  1 +
 meson.build                           |  3 +++
 meson_options.txt                     |  1 +
 6 files changed, 36 insertions(+)
 create mode 100644 gen/com/axiado/ax200/Demo/meson.build
 create mode 100644 gen/com/axiado/ax200/meson.build
 create mode 100644 gen/com/axiado/meson.build

diff --git a/gen/com/axiado/ax200/Demo/meson.build b/gen/com/axiado/ax200/Demo/meson.build
new file mode 100644
index 0000000..f879ccb
--- /dev/null
+++ b/gen/com/axiado/ax200/Demo/meson.build
@@ -0,0 +1,14 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/axiado/ax200/Demo__cpp'.underscorify(),
+    input: [ '../../../../../yaml/com/axiado/ax200/Demo.interface.yaml',  ],
+    output: [ 'server.cpp', 'server.hpp', 'client.hpp',  ],
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'com/axiado/ax200/Demo',
+    ],
+)
+
diff --git a/gen/com/axiado/ax200/meson.build b/gen/com/axiado/ax200/meson.build
new file mode 100644
index 0000000..cb3f78b
--- /dev/null
+++ b/gen/com/axiado/ax200/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+subdir('Demo')
+generated_others += custom_target(
+    'com/axiado/ax200/Demo__markdown'.underscorify(),
+    input: [ '../../../../yaml/com/axiado/ax200/Demo.interface.yaml',  ],
+    output: [ 'Demo.md' ],
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/axiado/ax200/Demo',
+    ],
+)
+
diff --git a/gen/com/axiado/meson.build b/gen/com/axiado/meson.build
new file mode 100644
index 0000000..a77baa8
--- /dev/null
+++ b/gen/com/axiado/meson.build
@@ -0,0 +1,2 @@
+# Generated file; do not modify.
+subdir('ax200')
diff --git a/gen/com/meson.build b/gen/com/meson.build
index 9a2e068..2f2783e 100644
--- a/gen/com/meson.build
+++ b/gen/com/meson.build
@@ -1,4 +1,5 @@
 # Generated file; do not modify.
+subdir('axiado')
 subdir('google')
 subdir('ibm')
 subdir('intel')
diff --git a/meson.build b/meson.build
index 8bba339..041c8c9 100644
--- a/meson.build
+++ b/meson.build
@@ -25,6 +25,9 @@ endif
 if get_option('data_com_google')
     selected_subdirs += 'com/google'
 endif
+if get_option('data_com_axiado')
+    selected_subdirs += 'com/axiado'
+endif
 if get_option('data_org_freedesktop')
     selected_subdirs += 'org/freedesktop'
 endif
diff --git a/meson_options.txt b/meson_options.txt
index eb92c99..fc91a02 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -3,6 +3,7 @@ option('libphosphor_dbus', type: 'boolean', value: true)
 option('data_com_ibm', type: 'boolean', value: true)
 option('data_com_intel', type: 'boolean', value: true)
 option('data_com_google', type: 'boolean', value: true)
+option('data_com_axiado', type: 'boolean', value: true)
 option('data_org_freedesktop', type: 'boolean', value: true)
 option('data_org_open_power', type: 'boolean', value: true)
 option('data_xyz_openbmc_project', type: 'boolean', value: true)
-- 
2.25.1

